packages/hebao_v1/contracts/modules/stores/WhitelistStore.sol

```diff
@@ -52,25 +60,37 @@ contract WhitelistStore is DataStore
         onlyManager
     {
         removeAddressFromSet(walletKey(wallet), addr);
+        delete timestampMap[wallet][addr];
         emit Whitelisted(wallet, addr, false);
     }
 
     function whitelist(address wallet)
         public
         view
-        returns (address[] memory)
+        returns (
+            address[] memory addresses,
+            uint[]    memory timestamps
+        )
     {
-        return addressesInSet(walletKey(wallet));
+        addresses = addressesInSet(walletKey(wallet));
+        timestamps = new uint[](addresses.length);
+        for (uint i = 0; i < addresses.length; i++) {
+            timestamps[i] = timestampMap[wallet][addresses[i]];
+        }
     }
 
     function isWhitelisted(
         address wallet,
         address addr)
         public
         view
-        returns (bool)
+        returns (
+            bool isWhitelistedAndEffective,
+            uint timestamp
+        )
     {
-        return isAddressInSet(walletKey(wallet), addr);
+        timestamp = timestampMap[wallet][addr];
+        isWhitelistedAndEffective = timestamp > 0 && timestamp <= now;
```

Comment:
I think this should be `isWhitelistedAndEffective = timestamp >= now;`

Reply:
The effectiveness timestamp is a time in the future when addToWhitelist is called, for exmaple:

```addToWhitelist(wallet, address, now + 1 day)```

. so the current logic is correct.

Reply:
Oh yes of course, for some reason I thought the check did the opposite.